### PR TITLE
[PORT] Adds in Alligators from Shiptest and adds in a Gator Crate

### PR DESCRIPTION
**Pull request by GraveHat**
Snapping turtles.

# Document the changes in your pull request

Adds in alligators from Shiptest, as well as the "pets" and the iguana found in shiptest-ss13/Shiptest#747. Also adds in a gator crate that can be bought for 3000 credits after the supply console has been emagged. Also adds Alligators to the fishing loot pool.

image
image

The gators have been adjusted to randomly decide to eat you like the goats and also drop 4 lizard meat and 1 lizard tail. They can be spawned as hostile mobs from gold extracts however their pet versions cannot.

# Spriting

image
image

image
image

image
image

# Wiki Documentation

Alligators Exist

# Changelog

:cl:  Gravehat, @swifterjack, @KillerOrcaCora for gator sprite
rscadd: Adds in an Iguana pet named Izzy, and a lesser gator pet named Steppy, no there is no general Iguana variant
rscadd: Adds alligators to the game
rscadd: Adds alligators to the fishing loot pool as a rare drop
rscadd: Adds in an alligator crate as a hidden crate
tweak: Alligators behave like goats and drop loot on butcher, can be spawned through gold slimes
imageadd: Adds gator sprites
/:cl:

### CODE REVIEW

**Comment by RG4ORDR** [comment]
gator hat gator hat gator hat gator hat gator hat gator hat gator hat gator hat gator hat gator hat gator hat

**Comment by JamieD1** [comment]
Why are there two different sizes in your pictures

**Review by cuackles — APPROVED** [review]

**Comment by GraveHat** [comment]
> Why are there two different sizes in your pictures

I used a fugu gland so we can see the sprites better.

File: code/modules/mob/living/simple_animal/friendly/gator.dm [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		if(enemies.len && prob(10))
+			enemies = list()
+			LoseTarget()
+			src.visible_message(span_notice("[src] calms down."))
+
+/mob/living/simple_animal/hostile/retaliate/gator/Retaliate()
+	. = ..()
+	src.visible_message(span_danger("[src] starts snapping [p_their()] jaws."))
+
+/mob/living/simple_animal/hostile/retaliate/gator/steppy
```
Comment (Bibby0110) [inline code/modules/mob/living/simple_animal/friendly/gator.dm:47]: remove the duplicate declarations

File: code/modules/mob/living/simple_animal/friendly/gator.dm [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	response_harm = "kicks"
+	gold_core_spawnable = NO_SPAWN
+	faction = list("neutral")
+	melee_damage_lower = 20
+	melee_damage_upper = 24
+	health = 125
+	maxHealth = 125
+	speed = 8
+
+/mob/living/simple_animal/hostile/retaliate/gator/steppy/iguana
```
Comment (Bibby0110) [inline code/modules/mob/living/simple_animal/friendly/gator.dm:56]: here too

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

**Review by ToasterBiome — DISMISSED** [review]
loot conflicts with https://github.com/yogstation13/Yogstation/pull/15486 can be added there instead

**Review by Bibby0110 — APPROVED** [review]

**Pull request merged**